linalg/strumpack.cpp

```diff
@@ -345,7 +334,11 @@ FactorInternal() const
    strumpack::ReturnCode ret = solver_->factor();
    if (ret != strumpack::ReturnCode::SUCCESS)
    {
+#if STRUMPACK_VERSION_MAJOR >= 6 && STRUMPACK_VERSION_MINOR >= 3 && STRUMPACK_VERSION_PATCH > 1
```

Comment:
This check is wrong: e.g. it will give false for v7.0.0 and v7.0.1.

I suspect there will be no new v6 version of STRUMPACK after v6.3.1, so the check can be simply
```c++
#if STRUMPACK_VERSION_MAJOR >= 7
```

Reply:
Whoops, good catch sorry about that. Working too quickly!